### PR TITLE
Whitelist module: blacklist functionalities

### PR DESCRIPTION
**Pull request by haythem96**
# Task: Feature Name:

Whitelist module: blacklist functionalities

## High Level Description

This PR change the following in the Whitelist module:
- Add blacklisting collateral, product and Otoken from owner.
- Refactored functions to allow for usage in both cases, whitelisting and blacklisting
- Remove require statement to check that owner not whitelisting an already whitelisted asset, less gas consumption for checks that does not harm the protocol.

### Code

- [x] Unit test 100% coverage
- [x] Does your code follow the naming and code documentation guidelines?

### Documentation

- [ ] Is your code up to date with the spec? 
- [ ] Have you added your tests to the testing doc?

### CODE REVIEW

File: contracts/Whitelist.sol [hunk 1]
```diff
@@ -105,60 +116,99 @@ contract Whitelist is Ownable {
...
      * @notice set an otoken address as whitelisted
      * @param _otokenAddress address of the oToken that is being whitelisted
+     * @param _state represent if a Otoken is whitelisted or not, true to whitelist and false to blacklist
      */
-    function _setWhitelistedOtoken(address _otokenAddress) internal {
-        require(!whitelistedOtoken[_otokenAddress], "Whitelist: Otoken already whitelisted");
-
-        whitelistedOtoken[_otokenAddress] = true;
+    function _setWhitelistedOtoken(address _otokenAddress, bool _state) internal {
+        whitelistedOtoken[_otokenAddress] = _state;
```
Comment (antoncoding) [inline contracts/Whitelist.sol:212]: I don't think we need this function anymore if it's just `whitelistedOtoken[_otokenAddress] = _state;`

**Review by antoncoding — APPROVED** [review]
LGTM 🔥

**Review by aparnakr — APPROVED** [review]
LGTM!

File: contracts/Whitelist.sol (hunk 1 above)
Comment (haythem96) [reply contracts/Whitelist.sol:212]: yeah agree, removed that.

**Pull request merged**